rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -325,14 +354,16 @@ public void setAllCached(String remotePath, long fileLength, long lastModified,
       return;
     }
     if (isInvalidationRequired(md.getLastModified(), lastModified)) {
-      invalidate(remotePath);
+      invalidateFileMetadata(remotePath);
```

Comment:
With new code, this will not delete the file now due to EXPLICIT eviction while we need the file to be deleted due to change in data

Reply:
nvm, I see it handled